src/cloud-api-adaptor/install/yamls/caa-pod.yaml

```diff
@@ -1,91 +0,0 @@
-apiVersion: apps/v1
-kind: DaemonSet
-metadata:
-  labels:
-    app: cloud-api-adaptor
-  name: cloud-api-adaptor-daemonset
-  namespace: confidential-containers-system
-spec:
-  selector:
-    matchLabels:
-      app: cloud-api-adaptor
-  updateStrategy:
-    type: RollingUpdate
-    rollingUpdate:
-      maxUnavailable: 1
-  template:
-    metadata:
-      labels:
-        app: cloud-api-adaptor
-    spec:
-      serviceAccountName: cloud-api-adaptor
-      containers:
-      - command:
-        - /usr/local/bin/entrypoint.sh
-        env:
-        - name: NODE_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.nodeName
-        envFrom:
-        - secretRef:
-            name: peer-pods-secret
-        - configMapRef:
-            name: peer-pods-cm
-        image: cloud-api-adaptor
-        imagePullPolicy: Always
-        name: cloud-api-adaptor-con
-        securityContext:
-          capabilities:
-            add:
-            - NET_ADMIN
-            - SYS_ADMIN
-        startupProbe:
-          httpGet:
-            path: /startup
-            port: 8000
-          failureThreshold: 30
-          periodSeconds: 20
-          initialDelaySeconds: 20
-        volumeMounts:
-        - name: auth-json
-          mountPath: "/root/containers/" # hardcoded
-          readOnly: true
-        - mountPath: /root/.ssh/
-          name: ssh
-          readOnly: true
-        - mountPath: /run/peerpod
-          name: pods-dir
-        - mountPath: /run/netns
-          mountPropagation: HostToContainer
-          name: netns
-        # # setting for cloud provider external plugin
```

Comment:
Hi @stevenhorsman !

Dynamically loading an external cloud provider was introduced by  @liudalibj in commit 549673b5.

I couldn't find a similar configuration on peer-pods's new controller in the operator, but I assume we can still set that settings via `kubectl edit`. Then I started worrying about this dynamically loading feature; in how many ways the new controller breaks it? :)

We might need to update the docs in src/cloud-api-adaptor/docs/addnewprovider.md too.

Reply:
Very good point - this is linked in with the concerns that Beraldo raised about the tls volume mounts - it's a bunch of stuff that with the switch to use the operator (so nothing to do with moving to kata `main`) has been broken and the user experience isn't properly thought about/defined :(